frontend/app/src/main/java/edu/pwr/iotmobile/androidimcs/ui/screens/admin/AdminViewModel.kt

```diff
@@ -41,4 +45,31 @@ class AdminViewModel : ViewModel() {
             )
         }
     }
+
+    fun deleteAccount(navigation: AdminNavigation) {
+        viewModelScope.launch {
+            kotlin.runCatching {
+                userRepository.deleteActiveUser()
+            }.onSuccess {
+                toast.toast("Deleted account")
+                navigation.openLogin()
+                return@launch
+            }.onFailure {
+                toast.toast("Failed deleting account")
+            }
+        }
```

Comment:
using kotlin.runCatching is like a try catch, just because the code is in onSuccess doesn't mean it succeeded, only that it didn't break